mllib/src/test/scala/org/apache/spark/ml/clustering/KMeansSuite.scala

```diff
@@ -108,6 +113,21 @@ class KMeansSuite extends SparkFunSuite with MLlibTestSparkContext with DefaultR
     val kmeans = new KMeans()
     testEstimatorAndModelReadWrite(kmeans, dataset, KMeansSuite.allParamSettings, checkModelData)
   }
+
+  test("Initialize using given cluster centers") {
+    val kmeans = new KMeans()
+      .setK(k)
+      .setSeed(1)
+      .setInitialModel(initialModel)
+    val model = kmeans.fit(dataset)
+
+    // Converged initial model should lead to only a single iteration.
+    val convergedModel = kmeans.setInitialModel(model).fit(dataset).clusterCenters
+    val oneIterationModel = kmeans.setInitialModel(model).setMaxIter(1).fit(dataset).clusterCenters
```

Comment:
Maybe what we should check it that with `setMaxIter` high values, the algorithm only use one iteration to converge. 

Also, nicer to have another test which is 

```
val almostConvergedModel = kmeans.setInitialModel(initialModel).setMaxIter(5).fit(dataset).clusterCenters
```

And then use it to train another k-means which should significantly reduce the # of iteration without warn start.